rir/src/compiler/opt/elide_env_spec.cpp

```diff
@@ -16,42 +16,96 @@ void ElideEnvSpec::apply(RirCompiler&, ClosureVersion* function,
 
     AvailableCheckpoints checkpoint(function, log);
 
+    auto nonObjectArgs = [&](Instruction* i) {
+        auto answer = true;
+        i->eachArg([&](Value* arg) {
+            if (arg->type.maybeObj() && arg->typeFeedback.maybeObj())
+                answer = true;
+        });
+        return answer;
+    };
+
+    std::function<Instruction*(Instruction*, BB*, bool)> nextInstr =
+        [&](Instruction* from, BB* bb, bool stepFirst) {
+            auto i = from;
+
+            if (stepFirst) {
+                auto ip = bb->atPosition(from);
+                if (++ip == bb->end()) {
+                    auto currentBB = bb->trueBranch();
+                    while (currentBB->isEmpty())
+                        currentBB = currentBB->trueBranch();
+                    ip = currentBB->begin();
+                }
+                i = *ip;
+            }
+
+            if (i->branches()) {
+                auto cp = Checkpoint::Cast(i);
+                assert(cp);
+                auto nextBB = bb->trueBranch();
+                while (nextBB->isEmpty())
+                    nextBB = nextBB->trueBranch();
+                return nextInstr(*(nextBB->begin()), nextBB, false);
```

Comment:
how can you know that the bb is not empty?

Reply:
Because the last two lines loop when the nextBB is empty?